tests/lean/run/6759.lean

```diff
@@ -0,0 +1,49 @@
+/-!
+# Match expressions with unnamed equality proofs
+
+https://github.com/leanprover/lean4/issues/6759
+
+Tests that `match` expressions successfully elaborate when providing a hole instead of an identifier
+to which to bind the optional equality proof.
+-/
+
+def getLast? {α : Type} (a : Array α) : Option α :=
+  match _ : a.size with
+  | 0 => none
+  | m + 1 => some a[m]
+
+/-- info: some 4 -/
+#guard_msgs in
+#eval getLast? #[3, 4]
+
+
+def getLasts? {α : Type} (a b : Array α) : Option (α × α) :=
+  match _ : a.size, _ : b.size with
+  | 0, _ => none
+  | _, 0 => none
+  | m + 1, n + 1 => some (a[m], b[n])
+
+/-- info: some (4, 5) -/
+#guard_msgs in
+#eval getLasts? #[3, 4] #[5]
+
+
+def getLasts?' (a b : Array α) : Option (α × α) :=
+  match _ : a.size, h : b.size with
+  | 0, _ => none
+  | _, 0 => none
+  | m + 1, n + 1 => some (a[m], b[n]'(h ▸ Nat.lt.base n))
+
+/-- info: some (4, 5) -/
+#guard_msgs in
+#eval getLasts?' #[3, 4] #[5]
+
+
+theorem test_tactic (n : Nat) (h : n - 1 > 2) : n - 1 > 2 := by
+  match _ : n - 1 with
```

Comment:
Would this work with ifs? Do they lower to match?

Reply:
To make sure I understand, are you asking whether Lean supports something like `if _ : some_decidable_prop then ... else ...`? If so, then yes, this was actually already supported; the purpose of this PR was to provide parity between `match` and `if` (see the example in #6759). Regarding whether `if` elaborates to a `match`: no, dependent `if` expressions elaborate to an application of `dite`, which is defined using the recursor for `Decidable` directly.